app/src/main/java/de/westnordost/streetcomplete/screens/settings/SettingsFragment.kt

```diff
@@ -261,4 +280,32 @@ class SettingsFragment :
             findPreference<Preference>("quest_presets")?.summary = summary
         }
     }
+
+    private fun setHiddenQuestsSummary() {
+        val amountOfHiddenQuests = countHiddenQuests()
+        viewLifecycleScope.launch {
+            val pref = findPreference<Preference>("quests.restore.hidden")
+            pref?.summary = calculateHiddenQuestsSummary(amountOfHiddenQuests)
+            if (amountOfHiddenQuests == 0L) {
+                pref?.isEnabled = false
+            }
+        }
```

Comment:
```suggestion
        val pref = findPreference<Preference>("quests.restore.hidden")
        viewLifecycleScope.launch {
            val amountOfHiddenQuests = withContext(Dispatchers.IO) { countHiddenQuests() }
            pref?.summary = calculateHiddenQuestsSummary(amountOfHiddenQuests)
            pref?.isEnabled = amountOfHiddenQuests > 0
        }
```

1. countHiddenQuests is an IO ooperation
2. preference needs to be enabled when amountOfHiddenQuests > 0